core/service/core.go

```diff
@@ -44,7 +58,10 @@ func StartServer() {
 
 	handler := handler.NewServiceHandler(userController, tournamentController)
 
-	var opts []grpc.ServerOption
+	opts := []grpc.ServerOption{
+		grpc.UnaryInterceptor(infrastructure.ErrorInterceptor(infrastructure.LogInterceptor)),
```

Comment:
I think infrastructure isn't a good naming for a package that contains grpc interceptors. I would rather call the package interceptor and the interceptor themselves Error and Log. So you could call them with something like interceptor.Error, interceptor.Log